app/renderer/bookmarks.js

```diff
@@ -0,0 +1,323 @@
+        bookmarkData.bookmarks.push(newbookmark)
+        bookmarkData.lastId += 1
+        writeBookmarksToFile()
+        notify(`Bookmark added: ${newbookmark.name.substring(0, 20)}:\
+                ${newbookmark.url.substring(0, 40)}`)
+    }
+}
+
+const fixBookmarkData = (option, value) => {
+    let correctFormat = ""
+    if (option === "path") {
+        if (!value[0] === "/") {
+            correctFormat = `/${value}`
+        } else {
+            correctFormat = value
+        }
+        correctFormat = correctFormat.replace(/\s/g, "-").replace(/[/]$/, "")
+    } else if (option === "tag") {
+        correctFormat = value.forEach(
+            t => t.replace(specialCharsAllowSpaces, ""))
+    } else if (option === "keywords") {
+        correctFormat = value.forEach(k => k.replace(specialChars, ""))
+    } else if (option === "name") {
+        correctFormat = value.replace(specialCharsAllowSpaces, "")
+    } else {
+        correctFormat = value
+    }
+    return correctFormat
+}
+
+const bookmarkObject = input => {
+    const newbookmark = {}
+    newbookmark.id = bookmarkData.lastId
+    // If there's "=" or "~" we parse the input.
+    if (input.join().includes("=") || input.join().includes("~")) {
+        // Retain spaces for multiple word inputs.
+        const options = input.join(" ").split("~")
+        for (let i = 0; i < options.length; i++) {
+            // Get key and value: [0,1]
+            const [key, value] = options[i].split("=")
+            if (key === "keywords" || key === "tag") {
+                const tagsOrKeywordList = value.split(",")
+                newbookmark[key] = tagsOrKeywordList
+            } else {
+                const allValue = options[i].split("=").slice(1).join("")
+                if (allValue?.trim()) {
+                    const correctData = fixBookmarkData(key,
+                        allValue)
+                    newbookmark[key] = correctData
+                }
+            }
+        }
+    } else {
+        // If there's only text, we take it as name for current page bookmark
+        newbookmark.name = input.join(" ")
+        if (!newbookmark.name?.trim()) {
+            // When user doesn't enter anything delete empty string
+            delete newbookmark.name
+        }
+    }
+    return newbookmark
+}
+
+const loadBookmark = input => {
+    const selectedBookmarks = matchBookmarksToInput(input)
+    const {navigateTo, addTab} = require("./tabs")
+    if (selectedBookmarks.length === 0) {
+        notify("Could not find bookmark.")
+    } else if (selectedBookmarks.length === 1) {
+        navigateTo(selectedBookmarks[0].url)
+    } else {
+        notify(`Loaded ${selectedBookmarks.length} bookmarks \
+                that matched the query.`)
+        selectedBookmarks.forEach(e => addTab({
+            "url": e.url
+        }))
+    }
+}
+
+const matchBookmarksToInput = input => {
+    const storedBookmarkData = getBookmarkData().bookmarks
```

Comment:
No need for calling the function in this same file, just access `bookmarkData` directly.